source/develop/release-management/features/virt/dedicated-cpu.md

```diff
@@ -183,6 +194,15 @@ containing `<cputune>` (with no arguments) with `<vcpupin>` elements. E.g.:
 </cputune>
 ```
 
+API call `VM.setNumberOfCpus` will be extended with an optional field
+`cpusets`. It is a list CPU lists where each item of the list describes pinning
+for the vCPU at that index. For example `['1','3','5']` means that vCPU `1`
+should be pinned to to pCPU `1`, vCPU `2` should be pinned to pCPU `3` and vCPU
+`3` should be pinned to pCPU `5`. While the field itself is technically
+optional it is required for VMs with policy other than `shared` or `pin`. When
+specified, length of the list must match the new number of CPUs (i.e. must
+match `numberOfCpus` field).
```

Comment:
it this a new representation that is different than that we use when starting/migrating a vm?

Reply:
Unfortunately yes. We may reconsider how we pass information for migration and use the same approach as we do here (instead of passing XML).

Reply:
Yes, this would be better. It's a bit unfortunate to use a different representation on VM startup and then on migrations and CPU hot plugs. But using XML for hot plugs would be weird and using different representations for migrations and CPU hot plugs would be messy and confusing.

Reply:
we do that (using xml) for hot-plug disk and hot-plug NIC - it's true that we don't have a definition of 'device' in this case but if we now pass the same content we have in the domain xml, what prevents us from using the same representation we have in the domain xml?

Reply:
Formally nothing. But the difference is that in case of devices, we can (at least in theory) to pass the provided XML snippet directly to libvirt. In case of pinning and CPU hot plug, the provided XML cannot be used, it must be parsed and the CPUs are pinned in individual libvirt calls using numeric arguments. That means we create and parse XML just for the sake of having XML in the API. It may be good for consistency, but it wastes resources and doesn't match the idea of direct communication with libvirt using domain XML snippets. So what do you think considering that, should we use XML here or not?